tools/testing/test_selections.py

```diff
@@ -1,6 +1,66 @@
-from typing import Dict, Tuple, List
+import csv
+import json
+import os
+import subprocess
 
-def calculate_shards(num_shards: int, tests: List[str], job_times: Dict[str, float]) -> List[Tuple[float, List[str]]]:
+from tools.stats.s3_stat_parser import (
+    get_previous_reports_for_branch,
+    get_previous_reports_for_pr,
+    Report, Version2Report,
+    HAVE_BOTO3)
+
+from typing import Any, Dict, List, Optional, Tuple, cast
+from typing_extensions import TypedDict
+
+class JobTimeJSON(TypedDict):
+    commit: str
+    JOB_BASE_NAME: str
+    job_times: Dict[str, float]
+
+
+def _get_stripped_CI_job() -> str:
+    """E.g. convert 'pytorch_windows_vs2019_py36_cuda10.1_build' to 'pytorch_windows_vs2019_py36_cuda10.1'.
+    """
+    job = os.environ.get("JOB_BASE_NAME", "").rstrip('0123456789')
+    if job.endswith('_slow_test'):
+        job = job[:len(job) - len('_slow_test')]
+    elif job.endswith('_test') or job.endswith('-test'):
+        job = job[:len(job) - len('_test')]
+    elif job.endswith('_build') or job.endswith('-build'):
+        job = job[:len(job) - len('_build')]
+    return job
+
+
+def _get_job_times_json(job_times: Dict[str, float]) -> JobTimeJSON:
+    return {
+        'commit': subprocess.check_output(['git', 'rev-parse', 'HEAD'], encoding="ascii").strip(),
+        'JOB_BASE_NAME': _get_stripped_CI_job(),
+        'job_times': job_times,
+    }
+
+
+def _calculate_job_times(reports: List["Report"]) -> Dict[str, float]:
```

Comment:
```suggestion
def calculate_job_times(reports: List["Report"]) -> Dict[str, float]:
```

Reply:
this was actually intended because `_calculate_job_times` is not exposed directly but instead via `export_s3_job_times` function.